lib/bloc_record/collection.rb

```diff
@@ -0,0 +1,38 @@
+module BlocRecord
+  class Collection < Array
+    def update_all(updates)
+      ids = self.map(&:id)
+      self.any? ? self.first.class.update(ids, updates) : false
+    end
+
+    def take(num=1)
```

Comment:
This needs to work on the current set. Your approach may add items that were not a part of the last query (and thus are not in this collection).